dbt/include/teradata/macros/adapters.sql

```diff
@@ -118,8 +118,23 @@
 {% endmacro %}
 
 {% macro teradata__get_columns_in_relation(relation) -%}
-    {% set use_qvci = var("use_qvci", "false") | as_bool %}
-     {{ log("use_qvci set to : " ~ use_qvci) }}
+    {% call statement('check_table_or_view', fetch_result=True) %}
+        SELECT TableKind FROM DBC.TablesV WHERE DatabaseName = '{{ relation.schema }}' AND TableName = '{{ relation.identifier }}'
+    {% endcall %}
+
+    {% set table_kind = load_result('check_table_or_view').table.columns['TableKind'].values()[0] %}
+
+    {%- if table_kind == 'V ' -%}
+        {% set temp_relation_for_view = relation.identifier ~ '_viw_tbl' %}
```

Comment:
can we use term `temp` label `temp_tbl` or 'temp_viw_tbl' to make it clear that its `temp`

Reply:
Sure, I'll make this change and append "tmp_viw_tbl" because dbt-core uses the "tmp" annotation for temporary tables.